scripts/zkai_attention_kv_stwo_native_d16_two_head_longseq_bounded_softmax_table_proof_input.py

```
@@ -0,0 +1,888 @@
+    for key, expected in expected_scalars.items():
+        if payload.get(key) != expected:
+            raise AttentionKvD16TwoHeadLongseqBoundedSoftmaxTableInputError(f"source payload {key} drift")
+    observed = sha256_hex_bytes(canonical_json_bytes(payload))
+    if observed != SOURCE_PAYLOAD_SHA256:
+        raise AttentionKvD16TwoHeadLongseqBoundedSoftmaxTableInputError("source payload commitment drift")
+    return payload
+
+
+def dot(lhs: list[int], rhs: list[int]) -> int:
+    if len(lhs) != KEY_WIDTH or len(rhs) != KEY_WIDTH:
+        raise AttentionKvD16TwoHeadLongseqBoundedSoftmaxTableInputError("dot-product width mismatch")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)